---
title: "Plotting haplotypes"
output: html_notebook
---

#packages loading
``` {r, echo=FALSE, results = FALSE, eval=FALSE}
# library(reticulate)
library(tidyverse)
library(ggplot2)
library(cowplot)
library(data.table)
# library(igraph)
# library("XML")a
library(xml2)
library(plyr)
library(dplyr)
# install.packages('ggbeeswarm')
# install.packages("extrafont") # see https://github.com/wch/extrafont/
# library(ggbeeswarm)


library(reshape2)

nice_layout<-   theme_cowplot(font_size = 22)+
    panel_border(color = "grey85", size = 1, linetype = 1,
  remove = FALSE, "black") +
    theme(strip.background = element_rect(fill = "#ffffff"), text = element_text(size = 22))


my_cols <- c('#a6cee3','#1f78b4','#b2df8a','#33a02c','#fb9a99','#e31a1c','#fdbf6f','#ff7f00','#cab2d6','#6a3d9a')


stackbarcolors<- c("ambiguous"="#808080", "caudatus"="#a6cee3", "cruentus"="#1f78b4", "hypochondriacus"="#b2df8a", "hybridus_CA"="#33a02c", "hybridus_SA"="#f8bf68", "quitensis"="#fb9a99")
```



## boxplot data preparation
```{r}
listagem=list()  # creates a empty list that will be appended as the funtion is applied
column_names<-c("pos", "caudatus", "cruentus", "hypochondriacus", "hybridus_CA", "hybridus_SA", "quitensis")
# especies<- c("cau", "cru", "hyp", "hyb", "qui")
# especies<- c( "caudatus", "cruentus", "hypochondriacus", "hybridus_CA", "hybridus_SA", "quitensis")
especies<-"caudatus"

for (chromosome in 16:16){  #cycles through each chromossome
for (species in especies){ #cycles per spcecies
for (i in 3:3) { #cicles per file == number of pop
      # ... make some data
  name_file <-paste0("/Volumes/T7/hap_data/sc",chromosome,"_haplotypes_",species,i)
    if(file.exists(name_file)) # checks file existance without breaking code in case of missing 
      dados<-fread(name_file)
    dados = dados[-1,]
    colnames(dados) <-column_names #subsitute first row for calculated columns
    dados.transformed<- dados %>%
      dplyr::mutate(donor=case_when( # checks if more than 50% of a site has a certain background
  cruentus > 0.5 ~ "cruentus",
    hypochondriacus > 0.5 ~ "hypochondriacus",
      caudatus > 0.5 ~ "caudatus",
      hybridus_CA > 0.5 ~ "hybridus_CA",
    hybridus_SA > 0.5 ~ "hybridus_SA",
    quitensis > 0.5 ~ "quitensis",
  TRUE   ~ "ambiguous" # when proability is over 1/2 (0.5)
)) %>%
  dplyr::group_by(donor) %>%  # groups by classified donors
  dplyr::summarise(contagem= n() /nrow(.))  #cpounts how many
                dados.transformed$recipient <-paste0(species)  #track of species of recipient 
       dados.transformed$file <-paste0(species,i)  #track of species of recipient and iteration number 
              dados.transformed$chr <-paste0(chromosome)  #track of species of recipient and iteration number 

      listagem<- rbind(listagem,dados.transformed)                   # Append new dataframe as the code iterates each haplotype (one chromossome per inidividual per population at a time)

}
}
}


```

# plot boxplot

```{r}

df1_long<-
melt(listagem, id.vars=c("recipient","donor"))

# listagem % >%
#   ggplot(aes(x=recipient, y=contagem)) +
#   geom_boxplot() +
#         ylim(0,1) + 
#   nice_layout

plot_side_by_side_boxplot<- df1_long %>%
  filter(variable=='contagem') %>%
  ggplot(aes(x=factor(recipient), y=as.numeric(value), fill=factor(donor))) +
  geom_boxplot(outlier.shape = NA) +
        ylim(0,1) +
    labs(fill='donor') +
  xlab("recipient") +
  ylab("proportion") +
  nice_layout +
  scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors) +  
    facet_grid(.~recipient, scales = "free") +
  # theme(strip.text.x = element_text(angle = 45, size =13)) +
                # theme(axis.text.x = element_blank())  
  theme(strip.text.x = element_blank()) +
  theme(axis.text.x=element_text(size=18)) +
theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(title = "Proportion distribution per population (chr 16)")




plot_side_by_side_boxplot
# 
ggsave( "../data/processed/figures/plot_side_by_side_boxplot_sc16.pdf", plot_side_by_side_boxplot, device="pdf", width = 12, height =8)



```

 ## boxplot data plotting by chromssome
 
```{r}

plot_boxplot_allchr<-listagem %>%
  ggplot(aes(x=chr, y=contagem )) +
      ylim(0,1) +
  geom_boxplot() +
        geom_jitter(size=0.4, alpha=0.9) +
      # geom_jitter(size=2, alpha=0.9, aes(color=as.factor(file))) + #only use when looking to single species
  facet_grid(donor~recipient, scales = "free") +
  nice_layout +
  ggtitle("108 samples") +
            theme(axis.text.x = element_text(angle = 90)) 
plot_boxplot_allchr
```




stackbarplot data preparation

```{r}

lista=list()
column_names<-c("pos", "caudatus", "cruentus", "hypochondriacus", "hybridus_CA", "hybridus_SA", "quitensis")
# especies<- c("cau", "cru", "hyp", "hyb", "qui")
# especies<- c( "caudatus", "cruentus", "hypochondriacus", "hybridus_CA", "hybridus_SA", "quitensis")
especies<-"cruentus"
subsample_chr_list<-(c(3:3)) #ubsample due to exhausted vector

for (chromosome in subsample_chr_list){
for (species in especies){
for (i in 5:5) {
  name_file <-paste0("/Volumes/T7/hap_data/sc",chromosome,"_haplotypes_",species,i)
    if(file.exists(name_file)) {
      dat<-fread(name_file)
    dat = dat[-1,]
    colnames(dat) <-column_names #subsitute first row for calcualted columns
    dat.transformed<- dat %>%
      dplyr::mutate(donor=case_when(
  cruentus > 0.5 ~ "cruentus",
    hypochondriacus > 0.5 ~ "hypochondriacus",
      caudatus > 0.5 ~ "caudatus",
      hybridus_CA > 0.5 ~ "hybridus_CA",
    hybridus_SA > 0.5 ~ "hybridus_SA",
    quitensis > 0.5 ~ "quitensis",
  TRUE   ~ "ambiguous" # when proability is over 1/2 (0.5)
)) 
        dat.transformed$recipient <-paste0(species)  #track of species of recipient 
        dat.transformed$file <-paste0(species,i)  #track of species of recipient and iteration number 
       dat.transformed$chr <-paste0(chromosome)  #track of species of recipient and iteration number 

      lista<- rbind(lista,dat.transformed)   
      # Append new dataframe

    }
else {print(paste0("The file does not exist, so it was skipped", name_file))
}
  }
}
}

options(scipen=999)



```

## table for Akansahka, summary by donoe

```{r}

# name_columns<- fread("/Users/josedias/Desktop/list_samples_names_transposed.txt", header=F)

scaffold_16_lista<-lista %>%
    select(-caudatus,-cruentus,-hypochondriacus,-quitensis,-hybridus_CA, -hybridus_SA, -recipient ) %>%
  group_by(file) %>%
      tidyr::spread(key = file,value = donor) 
colnames(scaffold_16_lista)<- name_columns

scaffold_8_lista %>%
fwrite(., "../data/processed/AS_files/AS_list_sc8.txt", quote=FALSE, sep="\t")



```

```{r}

ggsave("/Users/josedias/Desktop/sumarios/summary_hypochondraicus_sc16.png",
scaffold_8_lista %>% 
  select(-chr) %>%
  # head(n=10000) %>%
  melt(id.vars=("pos"))  %>%
  arrange(pos) %>%
  group_by(pos) %>%
  dplyr::summarise(count(value)) %>%
  mutate(prop=freq/21) %>%
        ggplot(.,aes(x=factor(pos), y=prop, fill=x)) +
      theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    #         ggplot(., aes(x = as.factor(Position), y = proportion, fill = x)) + 
          geom_bar( stat='identity',  width=1)  +
          nice_layout +
     scale_fill_manual(values = stackbarcolors),
height =10, width =28 )

```


create summaries of donors (per recipient)
```{r}
species<-"hybridus_SA"
subsample_chr_list<-(c(9:16)) #ubsample due to exhausted vector
lista=list()
sumario=list()
donor= c("ambiguous", "caudatus", "cruentus", "hybridus_CA", "hybridus_SA", "hypochondriacus", "quitensis" )
sumario<-as.data.frame(donor)
dat.transformed=list()
for (i in 1:5) {
  for (chromosome in subsample_chr_list){
  name_file <-paste0("/Volumes/T7/hap_data/sc",chromosome,"_haplotypes_",species,i)
    if(file.exists(name_file)) {
      dat<-fread(name_file)
    dat = dat[-1,]
    colnames(dat) <-column_names #subsitute first row for calcualted columns
    dat.transformed<- dat %>%
      dplyr::mutate(donor=case_when(
  cruentus > 0.5 ~ "cruentus",
    hypochondriacus > 0.5 ~ "hypochondriacus",
      caudatus > 0.5 ~ "caudatus",
      hybridus_CA > 0.5 ~ "hybridus_CA",
    hybridus_SA > 0.5 ~ "hybridus_SA",
    quitensis > 0.5 ~ "quitensis",
  TRUE   ~ "ambiguous" # when proability is over 1/2 (0.5)
)) 
        dat.transformed$recipient <-paste0(species)  #track of species of recipient 
        dat.transformed$file <-paste0(species,i)  #track of species of recipient and iteration number 
       dat.transformed$chr <-paste0(chromosome)  #track of species of recipient and iteration number 
      lista<- rbind(lista,dat.transformed)   

# sumario<-sumario %>%
#   inner_join(lista, by="donor")
# # dat.transformed<-list()
# lista<-cbind(lista,dat.transformed)
      # Append new dataframe

    }
else {print(paste0("The file does not exist, so it was skipped", name_file))
}

}
  test_lista<-
      lista %>%
      select(-caudatus,-cruentus,-hypochondriacus,-quitensis,-hybridus_CA, -hybridus_SA ) %>%
    dplyr::summarise(count(donor))  %>%
  mutate(prop=freq/sum(freq)) %>%
        select(-freq)
 colnames(test_lista)<-c("donor", paste0(species,i))
 
sumario<-sumario %>%
  inner_join(test_lista, by="donor")

}
# fwrite(sumario, "/Users/josedias/Desktop/sumarios/fixed_full_sumario_hybridus_SA.txt", sep="\t", quote=F)


# 
#   # lista<-
# 
# # colnames(lista)<-c("donor", paste0(species,i))â„¢
# sumario<-sumario %>%
#   inner_join(lista, by="donor")
# # dat.transformed<-list()
# lista<-cbind(lista,dat.transformed)
      # Append new dataframe

# fwrite(lista, "/Users/josedias/Desktop/sumario_hypochondriacus_all.txt", sep="\t", quote=F)
```

# plot summaries by donors
```{r}

sumario_cau<-fread("/Users/josedias/Desktop/sumarios/fixed_full_sumario_caudatus.txt") %>%
melt(., id.vars=c("donor")) %>% 
  mutate(recipient="caudatus")

sumario_cru<-fread("/Users/josedias/Desktop/sumarios/fixed_full_sumario_cruentus.txt") %>%
melt(., id.vars=c("donor")) %>% 
  mutate(recipient="cruentus")
sumario_hyp<-fread("/Users/josedias/Desktop/sumarios/fixed_all_full_sumario_hypochondriacus.txt") %>%
melt(., id.vars=c("donor")) %>% 
  mutate(recipient="hypochondriacus")

sumario_hybca<-fread("/Users/josedias/Desktop/sumarios/fixed_full_sumario_hybridus_CA.txt") %>%
melt(., id.vars=c("donor")) %>% 
  mutate(recipient="hybridus_CA")

sumario_hybsa<-fread("/Users/josedias/Desktop/sumarios/fixed_full_sumario_hybridus_SA.txt") %>%
melt(., id.vars=c("donor")) %>% 
  mutate(recipient="hybridus_SA")

sumario_qui<-fread("/Users/josedias/Desktop/sumarios/fixed_full_sumario_quitensis.txt") %>%
melt(., id.vars=c("donor"))%>% 
  mutate(recipient="quitensis")


whole_sumario<- rbind(sumario_cau,sumario_cru,sumario_hyp,sumario_hybca,sumario_hybsa,sumario_qui)

stackbarcolors<- c("ambiguous"="#808080", "caudatus"="#a6cee3", "cruentus"="#1f78b4", "hypochondriacus"="#b2df8a", "hybridus_CA"="#33a02c", "hybridus_SA"="#f8bf68", "quitensis"="#fb9a99")


ggplot() +
  geom_boxplot(data = sumario_cau, aes(x=donor, y=value, fill="#a6cee3", color="#a6cee3")) +
    geom_boxplot(data = sumario_cru, aes(x=donor, y=value, fill="#1f78b4", color="#1f78b4")) +
      geom_boxplot(data = sumario_hyp, aes(x=donor, y=value, fill="#b2df8a", color="#b2df8a")) +
      geom_boxplot(data = sumario_hybca, aes(x=donor, y=value, fill="#33a02c", color="#33a02c")) +
        geom_boxplot(data = sumario_hybsa, aes(x=donor, y=value, fill="#f8bf68", color="#f8bf68")) +
  geom_boxplot(data = sumario_qui, aes(x=donor, y=value, fill="#fb9a99", color="#fb9a99")) +
  scale_fill_discrete(labels=c('caudatus', 'cruentus', 'hypochondraicus', "hybridus_CA", "hybridus_SA", "quitensis")) +
    scale_color_discrete("recipient", labels=c('caudatus', 'cruentus', 'hypochondraicus', "hybridus_CA", "hybridus_SA", "quitensis")) +
  guides(fill = FALSE) +
  theme_bw()

ggsave("/Users/josedias/Desktop/sumarios/summary_donors+beeswarm.png",
ggplot() +
  # geom_boxplot(data = whole_sumario, aes(x=donor, y=value, fill=donor)) +
   geom_beeswarm(data = whole_sumario, aes(x=donor, y=value, color=donor, cex=1.5)) +
  geom_quasirandom(method = "pseudorandom") +
  facet_wrap(recipient~.) +
    facet_grid(.~recipient) +
  theme_bw() +
  ylab("proportion") +
   # scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors) +  
  nice_layout +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          axis.title.x = element_blank(),
          legend.box = "horizontal", legend.position = "none"),
height =10, width =28 )

```



```{r}

# ggsave("../data/processed/figures/hypochondriacus_21_proportionsummary_chromosome_sc9.png",
       ggsave("/Users/josedias/Desktop/cruentus_5_proportionsummary_chromosome_sc3.png",

  lista %>%
    # filter(pos<100000) %>%
            mutate(Position = pos/1e3) %>%
    select(-donor, -recipient, -file, -chr, -pos) %>%
         group_by(Position) %>%
    melt(, id.vars=c("Position"))  %>% 
# summarise(n_distinct(variable))
      ggplot(.,aes(x=as.factor(Position), y=value, fill=variable)) +
    #         ggplot(., aes(x = as.factor(Position), y = proportion, fill = x)) + 
  geom_bar(stat='identity', width = 1, na.rm = T) +
          # geom_bar( stat='identity',  width=1)  +
    # ylim(0,1) +
    ggtitle("hypochondriacus 643058 sc3") +
  ylab("proportion") +
      nice_layout +
          scale_color_manual(values = stackbarcolors) +
            scale_fill_manual(values = stackbarcolors) +
  theme(legend.position = "bottom", axis.text.x = element_blank()) +
    xlab("Position") ,
height = 8, width =18)

    scale_x_discrete(breaks = c("4000", "8000", "12000","16000", "21000")) 
  scale_x_discrete(name="position", expand = waiver(), guide = waiver())

everyother <- function(x) x[seq_along(as.numeric(x)) %% 100000 == 0]

  
    # scale_x_discrete(breaks = levels(Position)[c(T, rep(F, 9))])

  scale_x_discrete(breaks=c("5","10","15","20"))
  
  max(lista$pos)

  # axis.title.x =element_blank(), axis.text.x = element_blank())

# 
```

plot individuals files (incompatibility project)

```{r}

fwrite(listagem, "../data/processed/summary_hypochondriacus21.txt", quote=F, sep="\t")
fwrite(lista, "../data/processed/summary_along_the_genome_all_hypochondriacus.txt", quote=F, sep="\t")



ggsave("../data/processed/figures/caudatus_3_proportionsummary_chromosome.png",
listagem%>%
  ggplot(.,aes(x=donor, y=contagem, fill=donor)) +
  geom_bar(stat="identity") +
   facet_grid(.~chr, scales="free") +
  nice_layout +
      theme(axis.text.x = element_blank(),
            axis.ticks = element_blank()) +
        scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors) +  
  ggtitle("caudatus3 PI490612 "), 
height = 8, width =18)

    theme(axis.text.x = element_text(angle = 45, hjust=1))
  # filter(chr=="16")
    
    
    
  # ggsave("../data/processed/figures/caudatus_3_lineplot_sc16.png",
lista %>%
    filter(pos<100000) %>%

  # filter(chr==13) %>%
  ggplot( ) +
  geom_line(aes(x = pos, y=caudatus, color="caudatus")) +
    geom_point(aes(x = pos, y=caudatus, color="caudatus"), size=1) +
    geom_line(aes(x = pos, y=cruentus, color="cruentus")) +
    geom_point(aes(x = pos, y=cruentus, color="cruentus"), size=1) +
      geom_line(aes(x = pos, y=hypochondriacus, color="hypochondriacus")) +
    geom_point(aes(x = pos, y=hypochondriacus, color="hypochondriacus"), size=1) +
        geom_line(aes(x = pos, y=hybridus_CA, color="hybridus_CA")) +
    geom_point(aes(x = pos, y=hybridus_CA, color="hybridus_CA"), size=1) +
        geom_line(aes(x = pos, y=hybridus_SA, color="hybridus_SA")) +
    geom_point(aes(x = pos, y=hybridus_SA, color="hybridus_SA"), size=1) +
     geom_line(aes(x = pos, y=quitensis, color="quitensis")) +
    geom_point(aes(x = pos, y=quitensis, color="quitensis"), size=1) +
  nice_layout +
          scale_color_manual(values = stackbarcolors) +  
  theme(legend.position = "bottom")
# height = 4, width =15)

  
lista %>%
  # filter(chr==13) %>%
  filter(pos<100000) %>%
  ggplot( ) +
  geom_path(aes(x = pos, y=caudatus, color="caudatus")) +
    geom_point(aes(x = pos, y=caudatus, color="caudatus"), size=1) +
    geom_path(aes(x = pos, y=cruentus, color="cruentus")) +
    geom_point(aes(x = pos, y=cruentus, color="cruentus"), size=1) +
      geom_path(aes(x = pos, y=hypochondriacus, color="hypochondriacus")) +
    geom_point(aes(x = pos, y=hypochondriacus, color="hypochondriacus"), size=1) +
        geom_path(aes(x = pos, y=hybridus_CA, color="hybridus_CA")) +
    geom_point(aes(x = pos, y=hybridus_CA, color="hybridus_CA"), size=1) +
        geom_path(aes(x = pos, y=hybridus_SA, color="hybridus_SA")) +
    geom_point(aes(x = pos, y=hybridus_SA, color="hybridus_SA"), size=1) +
     geom_path(aes(x = pos, y=quitensis, color="quitensis")) +
    geom_point(aes(x = pos, y=quitensis, color="quitensis"), size=1) +
  nice_layout +
          scale_color_manual(values = stackbarcolors) +  
  theme(legend.position = "bottom")  



lista %>%
    filter(pos<100000) %>%

  # filter(chr==13) %>%
  ggplot( ) +
  geom_line(aes(x = pos, y=caudatus, color="caudatus")) +
    geom_point(aes(x = pos, y=caudatus, color="caudatus"), size=1) +
    geom_line(aes(x = pos, y=cruentus, color="cruentus")) +
    geom_point(aes(x = pos, y=cruentus, color="cruentus"), size=1) +
      geom_line(aes(x = pos, y=hypochondriacus, color="hypochondriacus")) +
    geom_point(aes(x = pos, y=hypochondriacus, color="hypochondriacus"), size=1) +
        geom_line(aes(x = pos, y=hybridus_CA, color="hybridus_CA")) +
    geom_point(aes(x = pos, y=hybridus_CA, color="hybridus_CA"), size=1) +
        geom_line(aes(x = pos, y=hybridus_SA, color="hybridus_SA")) +
    geom_point(aes(x = pos, y=hybridus_SA, color="hybridus_SA"), size=1) +
  nice_layout +
          scale_color_manual(values = stackbarcolors) +  
  theme(legend.position = "bottom")
# height = 4, width =15)

  # ggsave("../data/processed/figures/hypocondriacus_21_smooth_chromosome_sc9.png",
  ggsave("/Users/josedias/Desktop/cruentus_5_smooth_chromosome_sc3.png",

lista %>%
  # filter(chr==13) %>%
  # filter(pos<1000000) %>%
  ggplot() +
  # geom_line() +
  geom_smooth( aes(x = as.numeric(pos), y=quitensis, color="quitensis")) +
    # geom_point(aes(x = pos, y=caudatus, color="caudatus"), size=1) +
    geom_smooth(aes(x = pos, y=cruentus, color="cruentus")) +
    # geom_point(aes(x = pos, y=cruentus, color="cruentus"), size=1) +
      geom_smooth(aes(x = pos, y=caudatus, color="caudatus")) +
    # geom_point(aes(x = pos, y=hypochondriacus, color="hypochondriacus"), size=1) +
        geom_smooth(aes(x = pos, y=hybridus_CA, color="hybridus_CA")) +
    # geom_point(aes(x = pos, y=hybridus_CA, color="hybridus_CA"), size=1) +
        geom_smooth(aes(x = pos, y=hybridus_SA, color="hybridus_SA")) +
    # geom_point(aes(x = pos, y=hybridus_SA, color="hybridus_SA"), size=1) +
  nice_layout +
          scale_color_manual(values = stackbarcolors) +  
  theme(legend.position = "bottom") ,
height = 8, width =18)


lista %>%
  # filter(chr==13) %>%
  filter(pos<100000) %>%
melt(, id.vars=c("recipient","donor", "pos", "file")) %>%
rename(proportion=value) %>%
  filter(pos==94515)

# nrow()
  # nrow()
# nrow()
# mutate(proportion=as.numeric(freq)/29664) 
    group_by(pos) %>%
  dplyr::summarise(count(donor))

  # ggplot() +
             ggplot(., aes(x = as.factor(pos), y = proportion, fill = donor)) + 
     ggtitle(paste0(species, " recipient scaffold_9")) +
        geom_bar( stat='identity',  width=1) +
  # # geom_line() +
  # geom_smooth( aes(x = as.numeric(pos), y=caudatus, color="caudatus")) +
  #   # geom_point(aes(x = pos, y=caudatus, color="caudatus"), size=1) +
  #   geom_smooth(aes(x = pos, y=cruentus, color="cruentus")) +
  #   # geom_point(aes(x = pos, y=cruentus, color="cruentus"), size=1) +
  #     geom_smooth(aes(x = pos, y=hypochondriacus, color="hypochondriacus")) +
  #   # geom_point(aes(x = pos, y=hypochondriacus, color="hypochondriacus"), size=1) +
  #       geom_smooth(aes(x = pos, y=hybridus_CA, color="hybridus_CA")) +
  #   # geom_point(aes(x = pos, y=hybridus_CA, color="hybridus_CA"), size=1) +
  #       geom_smooth(aes(x = pos, y=hybridus_SA, color="hybridus_SA")) +
  #   # geom_point(aes(x = pos, y=hybridus_SA, color="hybridus_SA"), size=1) +

  nice_layout +
          scale_fill_manual(values = stackbarcolors) +  
  theme(legend.position = "bottom")  



  lista_caudatus3_sc9<-lista
  # ggsave("../data/processed/figures/caudatus_3_stackplot_sc16.png",
    ggsave("/Users/josedias/Desktop/caudatus_3_stackplot_sc9.png",
  lista_caudatus3_sc9 %>%
    # filter(pos<100000) %>%
            mutate(Position = pos/1e6) %>%
    select(-donor, -recipient, -file, -chr, -pos) %>%
         group_by(Position) %>%
    melt(, id.vars=c("Position")) %>%
# 
#    dplyr::summarise(count(donor)) %>%
#   mutate(donor=x) %>%
    ggplot(.,aes(x=as.numeric(Position), y=value, fill=variable)) +
    #         ggplot(., aes(x = as.factor(Position), y = proportion, fill = x)) + 
          geom_bar( stat='identity',  width=1)  +
    ggtitle("caudatus3 PI490612 sc9") +
      nice_layout +
          scale_color_manual(values = stackbarcolors) +
  theme(legend.position = "bottom", axis.title.x =element_blank(), axis.text.x = element_blank()),
height = 4, width =15) 

    
    
    
lista_caudatus3_sc9
```

# plot stackbarplot side-by-side proportions

```{r}

proportions=list()

# for (i in especies){
#  proportions<- }
   
   lista %>%
    dplyr::filter(recipient=="quitensis")
   nrow()


 proportions<- lista %>%
    filter(recipient=="hybridus_SA") %>%
# nrow()
    group_by(recipient) %>%
  dplyr::summarise(count(donor))  %>%

mutate(proportion=freq/4034324) %>%
  mutate(donor=x) 
 
  plot_proportions<- ggplot(proportions, aes(x = recipient, y = proportion, fill = donor)) + 
     ggtitle("proportions, chr 3") +
        geom_bar( stat='identity') +
    xlab("recipient") +
      scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors) +  
  nice_layout +
      theme(axis.text.x=element_text(size=18)) +
theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) 
        
  plot_proportions
  
  
  
  
      
  ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/plot_proportions_stackbar_side_by_side.pdf",plot_proportions, width = 14, height=9)


```



# plot stackbarplot along the genome

```{r}
especies<- c("hybridus_SA")

for (species in especies){
  nam <- paste( species,  "plot_per_site_sc3", sep = "_")
assign(nam, lista %>% 
  filter(recipient==especies) %>%
      # filter(pos<=14261388)  %>%
      #    filter(pos>= 13260388) %>%
      group_by(pos) %>%
   dplyr::summarise(count(donor)) %>%
  mutate(donor=x) %>%
  mutate(proportion=freq/5) %>%
    mutate(Position = pos/1e6))
         # filter(pos<=14261388)  %>%
         # filter(pos>= 14260388))
  #         ggplot(., aes(x = as.factor(Position), y = proportion, fill = x)) + 
  #    ggtitle(paste0(species,"recipient, chr 16")) +
  #       geom_bar( stat='identity') +
  # nice_layout +
  #         theme(axis.text.x = element_text(angle = 90)) +
  # theme(axis.title = element_blank()) +
  #             theme(axis.text.x = element_blank()) +
  #   xlab("Position")

}

```

## stackbarplot along the genome

```{r}

plot_to_save<- hybridus_SA_plot_per_site_sc3 %>%
         # filter(pos<=14261388)  %>%
         # filter(pos>= 14201388) %>%
            ggplot(., aes(x = as.factor(Position), y = proportion, fill = donor)) + 
     ggtitle(paste0(species, " recipient scaffold_9")) +
        geom_bar( stat='identity',  width=1) +
  # theme(axis.title = element_blank()) +
              # theme(axis.text.x = element_blank()) +
    xlab("Position (Mb)")    + 
scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors)  +
  nice_layout +
                theme(axis.text.x = element_blank()) +
        theme(axis.text.x = element_text(angle = 90)) +
    theme( # remove the vertical grid lines
           panel.grid.major.x = element_blank() ,
           # explicitly set the horizontal lines (or they will disappear too)
           panel.grid.major.y = element_line( size=.1, color="black" ) 
    ) +
                theme(axis.text.x = element_blank()) +
  ylim(0,1) +
 scale_x_discrete(labels=function(x) sprintf("%.0f", as.numeric(x))) 


plot_to_save
```


# stackbarplot QTL sc 16

```{r}

plot_to_save_QTL16<- quitensis_plot_per_site_sc16 %>%
   filter(pos>=900000)  %>%
         filter(pos<= 1050000) %>%
    mutate(Position_kb = pos/1e3) %>%
     ggplot(., aes(x = as.factor(Position_kb), y = proportion, fill = donor)) + 
     ggtitle(paste0(species, " recipient scaffold_16")) +
        geom_bar( stat='identity',  width=1) +
  # theme(axis.title = element_blank()) +
              # theme(axis.text.x = element_blank()) +
    xlab("Position (kb)")    + 
scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors)  +
  nice_layout +
                theme(axis.text.x = element_blank()) +
        # theme(axis.text.x = element_text(angle = 90)) +
                theme(axis.text.x = element_blank()) +
                theme(axis.ticks.x = element_blank()) +
    ylim(0,1) +
 scale_x_discrete(labels=function(x) sprintf("%.0f", as.numeric(x))) +
      geom_vline(xintercept=976,linetype='dashed', color='red', size=0.5) +
    geom_vline(xintercept=989, linetype='dashed', color='red', size=0.5) 
  
plot_to_save_QTL16

```



# line plot full genome

```{r}

line_plot_to_save<- hybridus_SA_plot_per_site_sc3 %>%
    ggplot(., aes(x =  Position, y = proportion, fill = donor, color=donor)) +
     ggtitle(paste0(species, " recipient scaffold_3")) +
        # geom_bar( stat='identity') +
          geom_line()  +
  geom_point(size=1) +
ylim(0,1) +
# theme(axis.title = element_blank()) +

    xlab("Position (Mb)")    + 
scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors)  +
  nice_layout +
                theme(axis.text.x = element_blank()) +
        theme(axis.text.x = element_text(angle = 90)) +
 scale_x_continuous(labels=function(x) sprintf("%.0f", as.numeric(x))) 
        # theme(legend.title=element_blank()) +
    # geom_rect(aes(xmin=976669, xmax=989943),   ymin = 0.0, ymax = 1, alpha = 0.2, fill=NA, colour="red") 
    # geom_vline(xintercept=976669,linetype='dashed', color='red', size=0.5) +
    # geom_vline(xintercept=989943, linetype='dashed', color='red', size=0.5) 

line_plot_to_save


```




 # line plot sc16 QTL  



```{r}

line_plot_to_save_QTL3<- hybridus_SA_plot_per_site_sc3 %>%
  filter(pos<5850000) %>%
  filter(pos>6000000) %>%
    mutate(Position_kb = pos/1e3) %>%
    ggplot(., aes(x =  Position_kb, y = proportion, fill = donor, color=donor)) +
     ggtitle(paste0(species, " recipient scaffold_9")) +
        # geom_bar( stat='identity') +
          geom_line()  +
  geom_point(size=1) +
ylim(0,1) +
# theme(axis.title = element_blank()) +
              # theme(axis.text.x = element_blank()) +
    xlab("Position (kb)")    + 
scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors)  +
  nice_layout +
                theme(axis.text.x = element_blank()) +
        theme(axis.text.x = element_text(angle = 90)) +
 scale_x_continuous(labels=function(x) sprintf("%.0f", as.numeric(x))) +
        # theme(legend.title=element_blank()) +
    # geom_rect(aes(xmin=976669, xmax=989943),   ymin = 0.0, ymax = 1, alpha = 0.2, fill=NA, colour="red") 
  geom_vline(xintercept= 5881.516,linetype='dashed', color='red', size=0.5) +
    geom_vline(xintercept=5966304, linetype='dashed', color='red', size=0.5) 


```

```{r}
# 
# 
#  ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/tmp_caudatus_plot_per_site_sc9.pdf",plot_to_save, width = 14, height=3)
# 
#  
#  ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/tmp_caudatus_plot_per_site_sc9.pdf",line_plot_to_save_QTL16, width = 14, height=3)
#  
# 
#  ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/quitensis_line_plot_per_site_sc16.pdf", line_plot_to_save, width = 14, height=3)
#  

 # ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/quitensis_line_plot_per_site_QTL_highlighted_sc16.pdf", line_plot_to_save_QTL_9, width = 14, height=3)
 
  # ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/tmp_caudatus_plot_per_site_sc9.pdf",plot_to_save, width = 14, height=3)

 
 # ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/tmp_caudatus_plot_per_site_QTL_highlighted_sc9.png",line_plot_to_save_QTL9, width = 14, height=3)
 

 ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/hybridus_SA_line_plot_per_site_sc3.png", line_plot_to_save, width = 14, height=3)
 

 ggsave(filename="/Users/josedias/mount/projects/ag-stetter/jdias/projects/geneflow/data/processed/figures/hybridus_SA_line_plot_per_site_QTL_highlighted_sc3.png", line_plot_to_save_QTL3, width = 14, height=3)
 
 
 
 # rm(plot_to_save)
# rm(caudatus_plot_per_site_sc16)
# rm(nam)

# 

```

# some lost code

```{r}


whole_genome_loc_inf<- fread("/Users/josedias/mount/projects/ag-stetter/jdias/projects/GeneFlow/data/processed/local_inference_whole_genome.txt")


plot_whole_genome_loc_inf_dodge <-
ggplot(whole_genome_loc_inf, aes(x=receiver, y=admixture, fill=donor)) +
  geom_bar(stat = "identity", position = "dodge") +
  nice_layout +
       scale_fill_manual(values = stackbarcolors) +
        scale_color_manual(values = stackbarcolors) 

ggsave("/Users/josedias/mount/projects/ag-stetter/jdias/projects/GeneFlow/data/processed/figures/plot_whole_genome_loc_inf_dodge.png", plot_whole_genome_loc_inf_dodge, width = 14, height=5, device = "png")
```



```{r}



fwrite(lista, "../data/processed/lista_fscp_ch3.txt", quote = F, sep ="\t")
fwrite(lista, "../data/processed/listagem_fscp_ch3.txt", quote = F, sep ="\t")


test_a<-c(1,2,3,4,5)
test_b<-c(1,2,3,4,5)

df<- data.frame(test_a,test_b)

df

ggplot(df, aes() )


```

